client/components/domains/register-domain-step/index.jsx

```diff
@@ -549,15 +549,14 @@ const RegisterDomainStep = React.createClass( {
 	showValidationErrorMessage: function( domain, error ) {
 		let message,
 			severity = 'error';
-		const lastIndexOfDot = domain.lastIndexOf( '.' ),
-			tld = lastIndexOfDot !== -1 && domain.substring( lastIndexOfDot ),
+		const tld = getTld( domain ),
```

Comment:
Ouch - it seems that `domain` here is actually `domainName` - it's a string with the domain name, not the domain object :/ Which, of course, throws an error in domain search (and domain mapping). All the more reason to switch to `domainName` for `getTld` :)